### PR TITLE
[WIP] Use human labels for wayfinding.

### PR DESCRIPTION
**Pull request by jmcarp**
* Add human labels to markup
* Use human labels for wayfinding text
* Drop `idToRef` and related helpers

Tasks remaining:
* [ ] Verify and implement behavior for appendices
* [ ] Ditto for interps

### CODE REVIEW

**Comment by jmcarp** [comment]
@cmc333333: merge conflicts are resolved. I'm not completely clear on how the wayfinding should work for appendices and interpretations, so it would be helpful if somebody could check this out and give feedback.

**Comment by cmc333333** [comment]
A few issues: this doesn't seem to "catch" when first loading a CFR section:
screen shot 2016-05-02 at 9 00 53 pm

It also has an issue when moving from the Preamble to CFR changes:
screen shot 2016-05-02 at 9 04 17 pm

I'm also running into a pretty significant issue where the header stops updating if I click on a nested ToC element. I bet this is related to eregs/notice-and-comment#158. To replicate,

1. click section III in the ToC of the preamble
2. click section III.C.1 in the ToC
3. scroll around
4. note that the header does not change

screen shot 2016-05-02 at 9 07 06 pm

On the bright side, appendices and interpretations look good:
screen shot 2016-05-02 at 9 11 25 pm
screen shot 2016-05-02 at 9 10 52 pm

**Comment by jmcarp** [comment]
@cmc333333: I pushed some fixes and minor refactors, and I think this is working as expected now.

File: regulations/static/regulations/js/source/views/main/child-view.js [hunk 1]
```diff
@@ -81,54 +81,42 @@ var ChildView = Backbone.View.extend({
...
+      // cache all sections in the DOM eligible to be the active section
+      // also cache some jQobjs that we will refer to frequently
+      this.$contentHeader = this.$contentHeader || $('header.reg-header');
+
+      // sections that are eligible for being the active section
+      this.$sections = this.$el
+        .find('li[id], .reg-section, .appendix-section, .supplement-section')
+        .map(function(idx, elm) {
+          return $(elm);
+        });
```
Comment (cmc333333) [inline regulations/static/regulations/js/source/views/main/child-view.js]: Minor, but it looks like this `map` converts a jQuery selection into a list? Any reason we need to make that conversion as opposed to leaving it a jQuery match?

Comment (jmcarp) [inline regulations/static/regulations/js/source/views/main/child-view.js]: The return value of the `map` should be a jQuery object, not a native array. Anyway, the purpose of the `map` is to cache the jQuerified elements--if we just use the match, iterating over the results gives us a list of native DOM elements, when our code wants jQuerified elements instead.

**Comment by cmc333333** [comment]
Getting closer. Two bugs:

Expanded CFR changes don't seem to scroll correctly:
screen shot 2016-05-09 at 3 18 42 pm. We can make this a separate issue.
---
The original CFR doesn't appear to be updating the header on AJAX loads:
1. Go to /447
2. Select the second section from the ToC
3. Scroll a bit
See:
screen shot 2016-05-09 at 3 21 29 pm

Similarly, if I hard refresh that page and then navigate elsewhere:
screen shot 2016-05-09 at 3 22 10 pm

This is bigger regression; I think we need to fix before we can merge.

**Comment by jmcarp** [comment]
@cmc333333: I might be misunderstanding the steps you described, but I'm seeing what I think are the correct section headers on scroll and ajax navigation. Can you tell me more about what you're seeing so I can replicate?

**Comment by cmc333333** [comment]
@jmcarp I must have had cached content; I'm only seeing the first issue (I'll make a different issue to resolve) in a private-viewing tab.

Very glad to see this merged finally.

**Pull request merged**